pex/wheel.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 Pex project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+from email.message import Message
+
+from pex.dist_metadata import MetadataFiles, MetadataType, load_metadata, parse_message
+from pex.typing import TYPE_CHECKING, cast
+
+if TYPE_CHECKING:
+    from typing import Dict, Text
+
+    import attr  # vendor:skip
+else:
+    from pex.third_party import attr
+
+
+@attr.s(frozen=True)
+class WHEEL(object):
+    """The .whl WHEEL metadata file.
+
+    See item 6 here for the WHEEL file contents: https://peps.python.org/pep-0427/#file-contents
+    """
+
+    class LoadError(Exception):
+        """Indicates an error loading WHEEL metadata."""
```

Comment:
It took some focus to distinguish `pex.pep_376.LoadError`, `pex.pep_427.WheelLoadError` and this new `WHEEL.LoadError`, especially when using two of them in the same file (as in pep_376.py, pep_427.py and installer.py).

AFAICT `pex.pep_376.LoadError` relates to loading the layout file for an InstalledWheel, which admittedly is slightly different, but the other two seem to mean the same thing (error loading the WHEEL metadata)?

So WDYT about consolidating at least the last two, and possibly distinguishing the other via naming, for example: 

`pex.pep_376.LoadError` -> `InstalledWheelLoadError`

Reply:
I went with `InstalledWheel.LoadError` for the one and moved `pex.pep_427.Wheel` to `pex.wheel.Wheel` and consolidated with `pex.wheel.WheelMetadataLoadError` now that `Wheel` and `WHEEL` are in the same module.